kvbc/include/categorization/base_types.h

```diff
@@ -43,23 +38,28 @@ struct KeyValueProof {
   std::vector<Hash> ordered_complement_kv_hashes;
 
   Hash calculateRootHash() const {
-    auto hasher = Hasher{};
-    const auto key_hash = hasher.digest(key.data(), key.size());
-    const auto value_hash = hasher.digest(value.data.data(), value.data.size());
+    const auto key_hash = Hasher{}.digest(key.data(), key.size());
+    const auto value_hash = Hasher{}.digest(value.data(), value.size());
 
-    // root_hash = h((h(k1) || h(v1)) || (h(k2) || h(v2)) || ... || (h(k3) || h(v3)))
+    // root_hash = h(h(k1) || h(v1) || h(k2) || h(v2) || ... || h(kn) || h(vn))
+    auto hasher = Hasher{};
     hasher.init();
+    const auto update = [&hasher](const auto &in) { hasher.update(in.data(), in.size()); };
+
     if (ordered_complement_kv_hashes.empty()) {
-      hasher.update(key_hash.data(), key_hash.size());
-      hasher.update(value_hash.data(), value_hash.size());
+      update(key_hash);
+      update(value_hash);
     } else {
       for (auto i = 0ul; i < ordered_complement_kv_hashes.size(); ++i) {
```

Comment:
nit: I think it's more readable to perform ```return hasher.finish()``` inside the if and return.
Then to remove the following ```else``` which makes it looks less important.